src/core/manager.c

```diff
@@ -464,14 +465,19 @@ static int manager_setup_timezone_change(Manager *m) {
          * Note that we create the new event source first here, before releasing the old one. This should optimize
          * behaviour as this way sd-event can reuse the old watch in case the inode didn't change. */
 
-        r = sd_event_add_inotify(m->event, &new_event, "/etc/localtime",
+        r = sd_event_add_inotify(m->event, &new_event, etc_localtime(),
                                  IN_ATTRIB|IN_MOVE_SELF|IN_CLOSE_WRITE|IN_DONT_FOLLOW, manager_dispatch_timezone_change, m);
         if (r == -ENOENT) {
                 /* If the file doesn't exist yet, subscribe to /etc instead, and wait until it is created either by
                  * O_CREATE or by rename() */
 
                 log_debug_errno(r, "/etc/localtime doesn't exist yet, watching /etc instead.");
-                r = sd_event_add_inotify(m->event, &new_event, "/etc",
+
+                r = path_extract_directory(etc_localtime(), &localtime_dir);
+                if (r < 0)
+                        return r;
+
+                r = sd_event_add_inotify(m->event, &new_event, localtime_dir,
```

Comment:
Unfortunately, the directory may neither exist. If the path `etc_localtime()` is configurable, then potentially we need to do something like `bus_socket_inotify_setup()`. That's quite ugly and complicated.

At least, please mention in ENVIRONMENT.md that the parent directory must always exist and must not be removed during runtime.

Reply:
please update the log_debug_errno() comment above, so that it outputs the righ tpath.